fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/source/FlussSourceBuilder.java

```diff
@@ -103,10 +106,8 @@ public FlussSourceBuilder<OUT> setDeserializationSchema(
         return this;
     }
 
-    // TODO: refactor this method to use field names instead of indexes
-    //  see: https://github.com/alibaba/fluss/issues/804
-    public FlussSourceBuilder<OUT> setProjectedFields(int[] projectedFields) {
-        this.projectedFields = projectedFields;
+    public FlussSourceBuilder<OUT> setProjectedFields(String[] projectedFieldNames) {
```

Comment:
We can use varargs here `setProjectedFields(String... projectedFieldNames)` to provide a more friendly API without constructing an array here. This avoids the need to manually construct an array and allows for cleaner usage in the Javadoc examples, `.setProjectedFields("orderId", "amount")`.